man/avg_showfont.1
.TH AVG_SHOWFONT "1" "March 2011" "libavg 1.5.4" "User Commands"
.SH NAME
avg_showfont \- font enumeration and preview tool
.SH SYNOPSIS
.B avg_showfont
[\fIfontname\fR] [\fItext\fR]
.SH DESCRIPTION
List available font faces and generate preview of their variants.
.PP
If invoked without arguments, avg_showfont generates a list of font family names and prints them on the console.
Fontconfig is used for font discovery. Fonts are searched for in the system font path and in $PWD/fonts. 
.PP
If \fBfontname\fR is specified, the chosen font is displayed in a window, with one line for
each available variant.
.PP
If supplied, the optional \fBtext\fR argument is used instead of the variant name in the
sample text.
.SH AUTHOR
Ulrich von Zadow <coder@c-base.org>


NEWS
libavg 1.5.4 - February 28, 2011 (Linux only)

  svn revision 5644, branches/release1.5.x

  XInput 2.1 updates/fixes.

libavg 1.5.3 - February 20, 2011 (Linux only)

  svn revision 5599

  Major new feature:
    * Linux XInput 2.1 Multitouch support.

libavg 1.5.1 - November 14, 2010 (Mac only)

  svn revision 5275

  Fixed bug that caused fullscreen apps to crash on the mac (Bug # 162).

libavg 1.5.0 - November 6, 2010

  svn revision 5270

  Major New Features:
    * Nodes can be constructed directly from python, making createNode unnecessary.
    * Added support for rendering avg scenes to offscreen canvases.
    * Added GPU-based effects framework and Blur, Shadow, Chromakey effects.
    * Added multitouch driver model and support for Apple Magic Trackpads and TUIO 
      devices.
    * Support for flash videos with an alpha channel. Decoding is GPU-accellerated.

  Other Changes:
    * Added color controls (gamma, brightness, contrast) to all RasterNodes.
    * Node constructors have a parent parameter which makes a following addChild 
      redundant.
    * Added Player.setEventHook().
    * Added onscreen keyboard for touchscreens.
    * WordsNode.getNumLines(), .getCharIndexFromPos(), .getTextAsDisplayed().
    * DivNode.getEffectiveMediaDir().
    * Clearer profiling that works for multiple threads.
    * Added AVG_BREAK_ON_IMPORT and AVG_BREAK_ON_ASSERT env variables.
    * Added AVG_DUMP_TEST_FRAMES env variable.
    * Added Player.setWindowFrame().
    * Added VideoNode.queuelength.
    * Video stability & timing improvements.
    * DivNode crop default is now false.
    * Node class names are unified: All names end with 'Node': WordsNode, DivNode, etc.
    * Added ImageNode.compression for 16 bpp texture support.
    * Added avg_audioplayer.py.

    * Various compiler compatibility fixes: Compiles with gcc 4.5 and VC++ 2010 now.
    * Lots of bugfixes - see the bug tracker.

libavg 1.5.0.pre3 - November 4, 2010

  svn revision 5267

libavg 1.5.0.pre2 - November 3, 2010

  svn revision 5263

libavg 1.5.0.pre1 - October 31, 2010

  svn revision 5244 

libavg 1.0.0 - January 8, 2010

  svn revision 4489

  New Features:

    * hinting is now optional in words node.
    * Multitouch emulation module added to AVGMTApp.
    * masks now work with words nodes.
    * Added video.volume property and video.hasAudio() method.
    * Added AVG_LOG_CATEGORIES environment variable. Log configuration is now
      done using the environment and not by calling a method.
    * Added maskpos and masksize attribute so masks can be moved and rescaled.
    * Added Player.getKeyModifierState()
    * Added sse2 version of yuv->rgb color conversion.
    * Added avg_videoinfo.sh.
    * Added mesh node.
    * Added avg_showcamera --resetbus to reset the firewire bus after crashes.

  Other Improvements:
    * New anim framework is now stable.
    * Snow Leopard compatibility.
    * avg_showcamera has much better command line handling.
    * The test suite is faster and a lot more comprehensive.
    * hrefs and mediadirs are now unicode-aware, so filenames like "ö.png" 
      should work.
    * node.unlink(True) now deletes all event handlers in the node, making
      it a lot easier to avoid memory leaks. 
    * Lots of bug fixes:
      - Windows gamma is now reset correctly on program end.
      - Linux vsync now much more stable.
      - Setting node.pos.x now throws an error instead of being ignored. 
        Same for all other Point2D attributes.
      - video.getNumFrames() now returns the exact number of frames.
      - Behaviour of nodes before Player.play() and when they are not in the
        avg tree is now much more stable.
      - Fixed words node sensitive area for centered and right-aligned nodes.

libavg 1.0.0.pre3 - January 6, 2010

  svn revision 4475

libavg 1.0.0.pre2 - December 23, 2009

  svn revision 4398

libavg 1.0.0.pre1 - December 21, 2009

  svn revision 4387

libavg 0.9.0 - August 23, 2009

  svn revision 4100

  Major new features:

    * Support for vector node types: Line, PolyLine, Curve, Circle, Polygon,
      Rectangle. Vector nodes can be textured, with the closed vectors taking two
      textures: one for the outline, one for the fill. Vector drawing is GPU-based.
    * Plugin support. Plugins written in C++ can define complete new node types.
    * Powerful (but still experimental) new animation framework.
    * Unified & much more stable camera drivers. Consistent support for all 
      resolutions & framerates we could test. Support for more than one camera on all
      platforms. fw800 is supported.
    * Uniform support for aggregate types in node interfaces (node.pos, node.pivot, 
      ...)

  Other improvements:
    * gcc 4.4 and ffmpeg 0.5.0 compatibility.
    * Added letterspacing support to words node.
    * Threaded videos are now the default.
    * Improved video seek and loop performance.
    * New Node::getElementByPos() method that returns the node at that point on the
      screen.
    * Added avg.svnrevision and avg.svndate properties so the avg version can be 
      queried.
    * New avg_showfont.py utility to help font selection.
    * New avg_showcamera.py utility to help camera setup.
    * New avg.getMemoryUsage function that returns the resident set size.
    * New Words.addFontDir().
    * Added support for mipmapping node textures, including videos.
    * rasternode.maskhref to support static alpha masks for any videos, word nodes,
      etc.

  Syntax and semantic changes:
    * In the course of adding aggregate types for node attributes, several attributes
      have changed:
      - words.size is now called words.fontsize.
      - pivotx and pivoty have been replaced by pivot.
    * right-aligned and centered words nodes are positioned differently.
    * avgtrackerrc syntax has changed significantly; see src/avgtrackerrc.minimal.
    * Support for firewire cameras using libdc1394 ver. 1 has been phased out. Please 

src/utils/avg_showfont.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
#
# libavg - Media Playback Engine.
# Copyright (C) 2003-2008 Ulrich von Zadow
#
# This library is free software; you can redistribute it and/or
# modify it under the terms of the GNU Lesser General Public
# License as published by the Free Software Foundation; either
# version 2 of the License, or (at your option) any later version.
#
# This library is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
# Lesser General Public License for more details.
#
# You should have received a copy of the GNU Lesser General Public
# License along with this library; if not, write to the Free Software
# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
#
# Current versions can be found at www.libavg.de
#

import sys
from libavg import avg
import time

Player = avg.Player.get()

Player.loadString("""
<?xml version="1.0"?>
<!DOCTYPE avg SYSTEM "../../doc/avg.dtd">
<avg width="640" height="480">
</avg>
""")
if len(sys.argv) ==1:
    print "Available fonts: "
    fontList = avg.WordsNode.getFontFamilies()
    print fontList
    print
    print "Usage: showfont.py <fontname> [<text>]"
    print
    sys.exit(1)
else:
    fontname=sys.argv[1]
    if len(sys.argv) > 2:
        displayText=sys.argv[2]
    else:
        displayText=""
variants = avg.WordsNode.getFontVariants(fontname)
print variants
rootNode = Player.getRootNode()
y = 10
for variant in variants:
    if displayText == "":
        text = fontname+": "+variant
    else:
        text = displayText
    node = Player.createNode("words", 
            { "text": text,
              "font": fontname,
              "variant": variant,
              "fontsize": 24,
              "x": 10,
              "y": y
            })
    rootNode.appendChild(node)
    y += 50
Player.setVBlankFramerate(1)
Player.play()


